examples/03_connectivity/plot_ica_neurovault.py

```diff
@@ -0,0 +1,148 @@
+"""
+NeuroVault cross-study ICA maps.
+================================
+
+This example shows how to download statistical maps from
+NeuroVault, label them with NeuroSynth terms,
+and compute ICA components across all the maps.
+
+See :func:`nilearn.datasets.fetch_neurovault`
+documentation for more details.
+
+"""
+# Author: Ben Cipollini
+# License: BSD
+# Ported from code authored by Chris Filo Gorgolewski, Gael Varoquaux
+# https://github.com/NeuroVault/neurovault_analysis
+import warnings
+
+import numpy as np
+from scipy import stats
+from sklearn.decomposition import FastICA
+
+from nilearn.datasets import fetch_neurovault
+from nilearn.image import new_img_like, load_img
+
+
+def clean_img(img, dtype=np.float32):
+    """ Remove nan/inf entries."""
+    img = load_img(img)
+    img_data = img.get_data().astype(dtype)
+    img_data[np.isnan(img_data)] = 0
+    img_data[np.isinf(img_data)] = 0
+    return new_img_like(img, img_data)
+
+
+######################################################################
+# Get image and term data
+
+# Download images
+# Here by default we only download 80 images to save time,
+# but for better results I recommend using at least 200.
+print("Fetching Neurovault images; "
+      "if you haven't downloaded any Neurovault data before "
+      "this will take several minutes.")
+nv_data = fetch_neurovault(max_images=80, fetch_neurosynth_words=True)
+
+images = nv_data['images']
+term_weights = nv_data['word_frequencies']
+vocabulary = nv_data['vocabulary']
+
+# Clean and report term scores
+term_weights[term_weights < 0] = 0
+total_scores = np.mean(term_weights, axis=0)
+
+print("\nTop 10 neurosynth terms from downloaded images:\n")
+print('{2:>20}{0:>15} : {1:<15}\n{2:>20}{2:->30}'.format(
+    'term', 'total score', ''))
+for term_idx in np.argsort(total_scores)[-10:][::-1]:
+    print('{0:>35} : {1:.3f}'.format(
+        vocabulary[term_idx], total_scores[term_idx]))
+
+
+######################################################################
+# Reshape and mask images
+
+from nilearn.datasets import load_mni152_brain_mask
+from nilearn.input_data import NiftiMasker
+
+print("\nReshaping and masking images.\n")
+
+with warnings.catch_warnings():
+    warnings.simplefilter('ignore', UserWarning)
+    warnings.simplefilter('ignore', DeprecationWarning)
+
+    mask_img = load_mni152_brain_mask()
+    masker = NiftiMasker(
+        mask_img=mask_img, memory='nilearn_cache', memory_level=1)
```

Comment:
Because people do not know what is `memory_level`, so they will look up the documentation, which is not very clear, and then they may end up thinking that this is too complicated.

Is a memory object is provided we should set memory level to 1 and not raise a warning.